docker/generic/start_proxy.py

```diff
@@ -394,7 +398,34 @@ def make_argparser():
         | /hello///        | Rejected              | /hello             |
         -----------------------------------------------------------------
         
-        By default, ESPv2 will not merge slashes.''')
+        By default, ESPv2 will merge slashes.
+        Disable the feature only if your traffic is affected by the behavior.
+        
+        For more details, see:
+        https://cloud.google.com/api-gateway/docs/path-templating''')
+
+    parser.add_argument('--escape_slashes_in_path', action='store_true',
+        help='''Enables escaping of the following characters in the `path`
+        HTTP header before the request is sent to the backend:
+        - %%2F or %%2f is converted to /
+        - %%5C or %%5c is converted to \
+        
+        When enabled, the behavior depends on the protocol:
+        - For OpenAPI backends, request paths with unescaped percent-encoded
+          slashes will be automatically escaped via a redirect.
+        - For gRPC backends, request paths with unescaped percent-encoded 
+          slashes will be rejected.
```

Comment:
qq: does it mean grpc backend will not be vulnerable?

Reply:
gRPC doesn't support redirects, so we just reject the request. That will prevent against the vulnerability occurring if the request is rejected.